fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -293,15 +457,82 @@ if [ ! -e "$fMRIFolder" ] ; then
+    ${FSLDIR}/bin/imrm "$fMRIFolder"/"$OrigTCSName"_prestc
+fi
 
-#Create fake "Scout" if it doesn't exist
-if [ $fMRIScout = "NONE" ] ; then
-  ${RUN} ${FSLDIR}/bin/fslroi "$fMRIFolder"/"$OrigTCSName" "$fMRIFolder"/"$OrigScoutName" 0 1
-else
-  cp "$fMRIScout" "$fMRIFolder"/"$OrigScoutName".nii.gz
+# --- Copy over scout (own or reference if specified), create fake if none exists
+
+if [ "$fMRIReference" != "NONE" ]; then
+    # --- copy over exisiting scout images
+    log_Msg "Copying Scout from Reference BOLD"
+    ${FSLDIR}/bin/imcp ${fMRIReferencePath}/Scout* ${fMRIFolder}
+
+    for simage in SBRef_nonlin SBRef_nonlin_norm
+    do
+        ${FSLDIR}/bin/imcp ${fMRIReferencePath}/"${fMRIReference}_${simage}" ${fMRIFolder}/"${NameOffMRI}_${simage}"
+    done
+
+    mkdir -p ${ResultsFolder}
+    ${FSLDIR}/bin/imcp ${ReferenceResultsFolder}/"${fMRIReference}_SBRef" ${ResultsFolder}/"${NameOffMRI}_SBRef"
+else    
+    # --- Create fake "Scout" if it doesn't exist
+    if [ $fMRIScout = "NONE" ] ; then
+      ${RUN} ${FSLDIR}/bin/fslroi "$fMRIFolder"/"$OrigTCSName" "$fMRIFolder"/"$OrigScoutName" 0 1
+    else
+      ${FSLDIR}/bin/imcp "$fMRIScout" "$fMRIFolder"/"$OrigScoutName"
+    fi
+fi
+
+if [ $DistortionCorrection = "NONE" ] ; then
+    # Processing is more robust to registration problems if the fMRI is in the same orientation as the
+    # standard template (MNI152) images, which can be accomplished using FSL's `fslreorient2std`.
+    # HOWEVER, if you reorient, other parameters (such as UnwarpDir) need to be adjusted accordingly.
+    # Rather than deal with those complications here, we limit reorienting to DistortionCorrection=NONE condition.
+
+    # First though, detect if reorienting is even necessary
+    xorient=`$FSLDIR/bin/fslval "$fMRIFolder"/"$OrigTCSName" qform_xorient | tr -d ' '`
+    yorient=`$FSLDIR/bin/fslval "$fMRIFolder"/"$OrigTCSName" qform_yorient | tr -d ' '`
+    zorient=`$FSLDIR/bin/fslval "$fMRIFolder"/"$OrigTCSName" qform_zorient | tr -d ' '`
+
+    log_Msg "$fMRIFolder/$OrigTCSName: xorient=${xorient}, yorient=${yorient}, zorient=${zorient}"
+
+    if [[ "$xorient" != "Right-to-Left" && "$xorient" != "Left-to-Right" || \
+          "$yorient" != "Posterior-to-Anterior" || \
+          "$zorient" != "Inferior-to-Superior" ]] ; then
+      reorient=TRUE
+    else 
+      reorient=FALSE
+    fi
+
+    if [ $reorient = "TRUE" ] ; then
+      log_Warn "Performing fslreorient2std! Please take that into account when using volume BOLD images in further analyses!"
+
+      # --- reorient BOLD
+      ${FSLDIR}/bin/immv "$fMRIFolder"/"$OrigTCSName" "$fMRIFolder"/"$OrigTCSName"_pre2std
+      ${FSLDIR}/bin/fslreorient2std "$fMRIFolder"/"$OrigTCSName"_pre2std "$fMRIFolder"/"$OrigTCSName"
+      ${FSLDIR}/bin/imrm "$fMRIFolder"/"$OrigTCSName"_pre2std
+
+      # --- reorient SCOUT
+      if [ "$fMRIReference" != "NONE" ]; then
```

Comment:
Why wrap the reorienting of the SCOUT in this conditional?

Reply:
It was an error, it should be:

```
if [ "$fMRIReference" = "NONE" ]; then
```

As we only want to reorient it when we are not using a copy of the reference scout, which has already been reoriented when processing the reference bold.